File: skills/uctoo-api-skill/scripts/api_client.js
const https = require('https');
const http = require('http');

class UctooApiClient {
    constructor() {
        this.baseUrl = process.env.BACKEND_URL || 'https://javatoarktsapi.uctoo.com';
        this.accessToken = null;
        this.tokenExpiration = null;
    }

    login(username, password) {
        return new Promise((resolve, reject) => {
            const endpoint = '/api/uctoo/auth/login';
            const body = JSON.stringify({ username, password });
            
            this.callApi(endpoint, 'POST', body, false)
                .then(result => {
                    try {
                        const json = JSON.parse(result);
                        if (json.data && json.data.access_token) {
                            this.accessToken = json.data.access_token;
                            this.tokenExpiration = Date.now() + 7200 * 1000;
                        }
                    } catch (e) {
                    }
                    resolve(result);
                })
                .catch(reject);
        });
    }

    isAuthenticated() {
        if (!this.accessToken) {
            return false;
        }
        if (this.tokenExpiration) {
            return Date.now() < this.tokenExpiration;
        }
        return true;
    }

    callApi(endpoint, method, body = '{}', requireAuth = true) {
        return new Promise((resolve, reject) => {
            let fullUrl = this.baseUrl;
            if (!this.baseUrl.endsWith('/') && !endpoint.startsWith('/')) {
                fullUrl += '/';
            }
            fullUrl += endpoint;

            const url = new URL(fullUrl);
            const isHttps = fullUrl.startsWith('https://');
            const client = isHttps ? https : http;

            const options = {
                hostname: url.hostname,
                port: url.port || (isHttps ? 443 : 80),
                path: url.pathname + url.search,
                method: method.toUpperCase(),
                headers: {
                    'Content-Type': 'application/json'
                }
            };

            if (requireAuth && this.isAuthenticated()) {
                options.headers['Authorization'] = `Bearer ${this.accessToken}`;
            }

            const req = client.request(options, (res) => {
                let data = '';

                res.on('data', (chunk) => {
                    data += chunk;
                });

                res.on('end', () => {
                    try {
                        JSON.parse(data);
                        resolve(data);
                    } catch (e) {
                        resolve(JSON.stringify({
                            error: 'API returned invalid JSON',
                            message: 'The API response is not in valid JSON format'
                        }));
                    }
                });
            });

            req.on('error', (e) => {
                resolve(JSON.stringify({
                    error: 'API call failed',
                    message: e.message
                }));
            });

            if (method.toUpperCase() !== 'GET' && body) {
                req.write(body);
            }

            req.end();
        });
    }

    getResource(entityType, id) {
        const endpoint = `/api/uctoo/${entityType}/${id}`;
        return this.callApi(endpoint, 'GET');
    }

    getResources(entityType, limit = 10, page = 0) {
        const endpoint = `/api/uctoo/${entityType}/${limit}/${page}`;
        return this.callApi(endpoint, 'GET');
    }

    createResource(entityType, data) {
        const endpoint = `/api/uctoo/${entityType}/add`;
        return this.callApi(endpoint, 'POST', typeof data === 'string' ? data : JSON.stringify(data));
    }

    updateResource(entityType, data) {
        const endpoint = `/api/uctoo/${entityType}/edit`;
        return this.callApi(endpoint, 'POST', typeof data === 'string' ? data : JSON.stringify(data));
    }

    deleteResource(entityType, id) {
        const endpoint = `/api/uctoo/${entityType}/del`;
        const body = JSON.stringify({ id });
        return this.callApi(endpoint, 'POST', body);
    }
}

if (require.main === module) {
    const client = new UctooApiClient();
    console.log('Uctoo API Client initialized');
    console.log('Base URL:', client.baseUrl);
}

module.exports = UctooApiClient;

File: skills/automation/notification/skill/scripts/api_client.py
#!/usr/bin/env python3
"""
UCTOO API Client Script
提供与 uctoo 后端 API 交互的功能
"""

import sys
import json
import requests
from typing import Optional, Dict, Any

BACKEND_URL = "https://javatoarktsapi.uctoo.com"


def make_request(
    method: str,
    endpoint: str,
    data: Optional[Dict[str, Any]] = None,
    token: Optional[str] = None
) -> str:
    """
    发送 HTTP 请求到 uctoo 后端 API
    
    参数:
        method: HTTP 方法 (GET, POST)
        endpoint: API 端点
        data: JSON 请求体
        token: 认证令牌
    
    返回:
        JSON 格式的响应字符串
    """
    url = f"{BACKEND_URL}{endpoint}"
    headers = {"Content-Type": "application/json"}
    
    if token:
        headers["Authorization"] = f"Bearer {token}"
    
    try:
        if method.upper() == "GET":
            response = requests.get(url, headers=headers, params=data)
        elif method.upper() == "POST":
            response = requests.post(url, headers=headers, json=data)
        else:
            return json.dumps({"error": f"Unsupported method: {method}"})
        
        return response.text
    except Exception as e:
        return json.dumps({"error": str(e)})


def main():
    """
    命令行入口
    
    用法: python api_client.py <method> <endpoint> [data_json] [token]
    """
    if len(sys.argv) < 3:
        print(json.dumps({"error": "Usage: python api_client.py <method> <endpoint> [data_json] [token]"}))
        sys.exit(1)
    
    method = sys.argv[1]
    endpoint = sys.argv[2]
    data = json.loads(sys.argv[3]) if len(sys.argv) > 3 else None
    token = sys.argv[4] if len(sys.argv) > 4 else None
    
    result = make_request(method, endpoint, data, token)
    print(result)


if __name__ == "__main__":
    main()
